packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -113,17 +120,24 @@ export const start = () => {
         WrappedApp: routeComponent(App, false, locals)
     }
 
-    return Promise.resolve()
-        .then(() => new Promise((resolve) => loadableReady(resolve)))
-        .then(() => {
-            hydrateRoot(
-                rootEl,
-                <OuterApp
-                    {...props}
-                    onHydrate={() => {
-                        window.__HYDRATING__ = false
-                    }}
-                />
-            )
-        })
+    await loadableReady()
+
+    const shouldHydrate = typeof disableSSR === 'boolean' ? disableSSR : !detectStorefrontPreview()
+
+    if (shouldHydrate) {
+        await hydrateRoot(
+            rootEl,
+            <OuterApp
+                {...props}
+                onHydrate={() => {
+                    window.__HYDRATING__ = false
+                }}
+            />
+        )
+    } else {
+        const root = createRoot(rootEl)
+        root.render(<OuterApp {...props} />)
+        // Clean up analogous to `onHydrate`
+        window.__HYDRATING__ = false
```

Comment:
Should this go before or after the render?

Reply:
I feel like it should be before! otherwise the initial render of the components may read `window.__HYDRATING__` incorrectly

Reply:
Looks like there are two places we use `isHydrated()` (`!window.__HYDRATING__`). PLP and header component:

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/8b8264edbd32e95958a53ddece5f81d36e95ae64/packages/template-retail-react-app/app/pages/product-list/index.jsx#L501

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/8b8264edbd32e95958a53ddece5f81d36e95ae64/packages/template-retail-react-app/app/components/header/index.jsx#L167

I'll double check the behavior of those two components with the flag disabled before/after rendering.

Reply:
Seems like there's not really any impact of before/after, at least on the dev server, so I went with before, since it's technically true.